### ISSUE TITLE
RRTMGP: t_int is never updated

### ISSUE THREAD

**Issue by AaronDonahue** [E3SM-Project/scream#1193]
RRTMGP is the only process that has the temperature at the interface as a "required" field.  However, it isn't updated anywhere, so currently t_int is initialized but quickly gets out-of-sync with t_mid since it is never updated.

Potential solution:
Switch `t_int` from a required field to a field that is calculated on the fly using ColumnOps

**Comment by AaronDonahue** [comment]
@bartgol and @brhillman , tagging you both because your feedback on how to remedy this is valuable.

**Comment by bartgol** [comment]
Yes, I think, for now, we could have rrtmgp compute T_int internally, without even storing it as a "Field". A raw view would do. And then compute it as a pre-process step using ColumnOps. This would, by the way, allow us to remove T_int from our init files.

**Comment by PeterCaldwell** [comment]
I really like the idea of keeping the FM as lean as possible by calculating things on the fly wherever we can. I have a vague recollection that there was some challenge to converting cell mean T to interface T that prevented us from doing this earlier... but since we need to convert all the cell-ave T changes from other procs to interface vals for radiation, I don't see how computing interface T on the fly could be any worse than what we're already doing.

**Comment by bartgol** [comment]
> I really like the idea of keeping the FM as lean as possible by calculating things on the fly wherever we can. I have a vague recollection that there was some challenge to converting cell mean T to interface T that prevented us from doing this earlier... but since we need to convert all the cell-ave T changes from other procs to interface vals for radiation, I don't see how computing interface T on the fly could be any worse than what we're already doing.

I agree. The mid->int issue was a long discussion. Since the number of use cases was _very limited_, we settled for not doing it.

The routine for mid->int interpolation that we have in ColumnOps is probably not great, since it is not monotonic (it can introduce new extrema). I originally implemented that one, because it was the inverse of int->mid (requiring one bc at top or bottom). Perhaps, the best thing to do here is a weighted average:

```
T_int (k) = (  T_int(k+1)*dz(k) + T_int(k)*dz(k+1)  ) / (dz(k) + dz(k+1))
```
with a bc both at top and bottom. It's simple enough, and does not introduce extrema.

**Comment by PeterCaldwell** [comment]
Yeah, weighted ave sounds good to me. I don't think we need a perfect implementation for radiation - it is using these numbers to perform calculations that themselves are imperfect, so imperfect T shouldn't kill us. It's not like we need to exactly conserve mass or something.

**Comment by AaronDonahue** [comment]
Excellent, so we should work on a branch which
1.  adds the weighted average to ColumnOps, and
2. changes the rrtmgp interface to no longer register t_int in FM, instead calculate on the fly.
3. and maybe remove t_int from our initial condition files.

**Comment by brhillman** [comment]
That all sounds good to me.

**Comment by tcclevenger** [comment]
PR https://github.com/E3SM-Project/scream/pull/1216 introduces the weighted avg into column-ops. Now I'll work on using it in rrtmgp.

**Comment by tcclevenger** [comment]
Yes, this has
